src/main/scala/ru/org/codingteam/keter/scenes/inventory/InventoryViewModel.scala

```diff
@@ -20,9 +21,45 @@ class InventoryViewModel(var inventory: Inventory) {
     promise.success(None)
   }
 
-  def backpackCategories = new ItemsViewModel(
-    ListMap(EquipmentCategory.All -> "ALL", EquipmentCategory.Weapon -> "Weapons")
-  )
-  def currentFolderItems = new ItemsViewModel[EquipmentItem](ListMap()) // TODO: Implement change logic
-  def currentItemInfo = new StaticTextViewModel("") // TODO: Implement logic
+  val backpackCategories = new ItemsViewModel(
+    VectorMap(All -> "ALL", Hat -> "Hats", Weapon -> "Weapons")
+  ) {
+
+    override protected def onSelectedItemChanged(item: Option[EquipmentCategory.Value]): Unit = {
+      val folderItems = item match {
+        case None => VectorMap[EquipmentItem, String]()
+        case Some(category) =>
+          val equipment = inventory.allEquipment.filter(e => category.contains(e.category))
+          toVectorMap(equipment.toSeq)
+      }
```

Comment:
Would this match look better if you replace it with `item.map(/* ... */).getOrElse(/* ... */)` pattern?